### PR TITLE
Use new method to register scroll effects

### PR DESCRIPTION
**Pull request by blasten**
Issue: #161 cc @keanulee @frankiefu

### CODE REVIEW

File: app-header/test/app-header.html [hunk 1]
```diff
@@ -119,9 +120,10 @@
           tearDown: sinon.spy(),
           run: sinon.spy()
         };
-        Polymer.AppLayout.scrollEffects = {
-          'test-effect': testEffect
-        };
+
+        effectCount++;
+        effectName = 'test-effect-' + effectCount;
```
Comment (keanulee) [inline app-header/test/app-header.html]: Why do we need `effectCount`? Can't we just have one effect across all tests?

File: app-scroll-effects/app-scroll-effects-behavior.html [hunk 2]
```diff
@@ -214,29 +215,28 @@
...
-     * @param {string} effectName The name of an effect defined in `Polymer.AppLayout.scrollEffects`
-     * @param {Object=} effectConfig The effect config object if the effect accepts config
-     *   values. (Optional)
-     * @return {Object} The effect object. This object has the following interface:
+     * @param {string} effectName The effect's name registered via `Polymer.AppLayout.registerEffect`.
+     * @param {Object=} effectConfig The effect config object. (Optional)
+     * @return {Object} An effect object with the following functions:
      *
-     *  * `obj.setUp()`, Sets up the requirements for the effect.
+     *  * `effect.setUp()`, Sets up the requirements for the effect.
```
Comment (keanulee) [inline app-scroll-effects/app-scroll-effects-behavior.html]: `effectConfig.setup()` would align it with the name of the parameter.

File: app-header/test/app-header.html (hunk 1 above)
Comment (blasten) [inline app-header/test/app-header.html]: Two reasons: 1. `setup` is ran multiple times and now `registerEffect` checks that the effect hasn't been registered yet (to help developers troubleshooting their apps).  2. there isn't a way to unregister effects yet bc I didn't find a use case for that.

File: app-scroll-effects/app-scroll-effects-behavior.html (hunk 2 above)
Comment (blasten) [inline app-scroll-effects/app-scroll-effects-behavior.html]: hmm. This isn't the `effectConfig` though. It's the value returned from `createEffect`.

Comment (keanulee) [inline app-scroll-effects/app-scroll-effects-behavior.html]: My bad, misread that

File: app-header/test/app-header.html (hunk 1 above)
Comment (keanulee) [inline app-header/test/app-header.html]: You could register the effect outside of `setup` and reset the spies:

```javascript

      var testEffect = {
        setUp: sinon.spy(),
        tearDown: sinon.spy(),
        run: sinon.spy()
      };
      var effectName = 'test-effect';
      Polymer.AppLayout.registerEffect(effectName, testEffect);

      setup(function() {
        container = fixture('testHeader');
        header = container.querySelector('app-header');
        toolbar = container.querySelector('app-toolbar');

        testEffect.setUp.reset();
        testEffect.tearDown.reset();
        testEffect.run.reset();
      });
```

Comment (blasten) [inline app-header/test/app-header.html]: True. That's another way to do it

File: app-toolbar/demo/index.html [hunk 3]
```diff
@@ -51,6 +51,14 @@
...
     }
 
+    
+
+    @media (max-width: 600px) {
+      app-toolbar {
+        height: 20px;
+      }
+    }
+
```
Comment (frankiefu) [inline app-toolbar/demo/index.html]: Is this change intentional?  If so, why do we need to do this?

Comment (blasten) [inline app-toolbar/demo/index.html]: 😓

**Comment by frankiefu** [comment]
We should have tests in app-scroll-effects.  Just file an issue: https://github.com/PolymerLabs/app-layout/issues/172

**Comment by frankiefu** [comment]
LGTM
